tests/conftest.py

```diff
@@ -484,8 +486,11 @@ def get_events(self, event_type=None):
             # filter heartbeat events to reduce noise
             if req.get("request_type") == "app-heartbeat" and self.filter_heartbeats:
                 continue
+            if (req["tracer_time"], req["seq_id"]) in self.gotten_events:
+                continue
             if event_type is None or req["request_type"] == event_type:
                 requests.append(req)
+                self.gotten_events.add((req["tracer_time"], req["seq_id"]))
```

Comment:
should `requests` just be a set, or a `dict`?

```python
key = (req["tracer_time"], req["seq_id"])
requests[key] = req

return sorted(requests.values())
```


this is bike shedding, basically the same result?

Reply:
Yeah it should!